backend/btrixcloud/colls.py

```diff
@@ -554,32 +560,24 @@ async def list_collections(
 
         return collections, total
 
+    # pylint: disable=too-many-locals
     async def get_collection_crawl_resources(
-        self, coll_id: UUID
+        self, coll_id: Optional[UUID], org: Organization
     ) -> tuple[List[CrawlFileOut], List[str], bool]:
         """Return pre-signed resources for all collection crawl files."""
-        # Ensure collection exists
-        _ = await self.get_collection_raw(coll_id)
+        match: dict[str, Any]
 
-        resources = []
-        pages_optimized = True
+        if coll_id:
+            crawl_ids = await self.get_collection_crawl_ids(coll_id)
+            match = {"_id": {"$in": crawl_ids}}
```

Comment:
```suggestion
            _ = await self.get_collection_raw(coll_id)
            crawl_ids = await self.get_collection_crawl_ids(coll_id)
            match = {"_id": {"$in": crawl_ids}}
```

Might be worth keeping this check in the if coll_id conditional below to avoid the subsequent queries and raise a 404 if the collection doesn't exist

Reply:
The places that call this already do call `get_collection_raw`, so thought it was redundant..

Reply:
Will be further addressed by collection org checking as well..